src/Zynga/Framework/StorableObject/Collections/Map/V2/Mock/StorableObjectMap.hh

```diff
@@ -0,0 +1,224 @@
+      array(
+        TestStorableObject::EXAMPLE_STRING_KEY => ConstKeys::STRING_VAL_1,
+        TestStorableObject::EXAMPLE_UINT64_KEY => ConstKeys::UINT_VAL_1,
+        TestStorableObject::EXAMPLE_FLOAT_KEY => ConstKeys::FLOAT_VAL_1,
+      ),
+      array(
+        TestStorableObject::EXAMPLE_STRING_KEY => ConstKeys::STRING_VAL_2,
+        TestStorableObject::EXAMPLE_UINT64_KEY => ConstKeys::UINT_VAL_2,
+        TestStorableObject::EXAMPLE_FLOAT_KEY => ConstKeys::FLOAT_VAL_2,
+      ),
+    );
+
+    return json_encode($arr);
+  }
+
+  public static function getKeyedJsonForNestedVector(): string {
+    $arr = array(
+      ConstKeys::INDEXED_KEY_0 => array(
+        TestStorableObject::EXAMPLE_STRING_KEY => ConstKeys::STRING_VAL_0,
+        TestStorableObject::EXAMPLE_UINT64_KEY => ConstKeys::UINT_VAL_0,
+        TestStorableObject::EXAMPLE_FLOAT_KEY => ConstKeys::FLOAT_VAL_0,
+      ),
+      ConstKeys::INDEXED_KEY_1 => array(
+        TestStorableObject::EXAMPLE_STRING_KEY => ConstKeys::STRING_VAL_1,
+        TestStorableObject::EXAMPLE_UINT64_KEY => ConstKeys::UINT_VAL_1,
+        TestStorableObject::EXAMPLE_FLOAT_KEY => ConstKeys::FLOAT_VAL_1,
+      ),
+      ConstKeys::INDEXED_KEY_2 => array(
+        TestStorableObject::EXAMPLE_STRING_KEY => ConstKeys::STRING_VAL_2,
+        TestStorableObject::EXAMPLE_UINT64_KEY => ConstKeys::UINT_VAL_2,
+        TestStorableObject::EXAMPLE_FLOAT_KEY => ConstKeys::FLOAT_VAL_2,
+      ),
+    );
+
+    return json_encode($arr, JSON_FORCE_OBJECT);
+  }
+
+  /**
+   * Helper functions for Map related data
+   */
+  public static function getRawNestedMap(): Map<string, mixed> {
+    $vector = self::getRawNestedVector();
+    return Map {
+      ConstKeys::STRING_KEY_0 => $vector->at(0),
+      ConstKeys::STRING_KEY_1 => $vector->at(1),
+      ConstKeys::STRING_KEY_2 => $vector->at(2),
+    };
+  }
+
+  public static function getPopulatedCollectionForNestedMap(
+  ): StorableMapCollection<TestStorableObject> {
+    $expected = new StorableMap(TestStorableObject::class);
+
+    $testObj0 = new TestStorableObject();
+    $testObj1 = new TestStorableObject();
+    $testObj2 = new TestStorableObject();
+
+    $testObj0->example_string->set(ConstKeys::STRING_VAL_0);
+    $testObj0->example_uint64->set(ConstKeys::UINT_VAL_0);
+    $testObj0->example_float->set(ConstKeys::FLOAT_VAL_0);
+
+    $testObj1->example_string->set(ConstKeys::STRING_VAL_1);
+    $testObj1->example_uint64->set(ConstKeys::UINT_VAL_1);
+    $testObj1->example_float->set(ConstKeys::FLOAT_VAL_1);
+
+    $testObj2->example_string->set(ConstKeys::STRING_VAL_2);
+    $testObj2->example_uint64->set(ConstKeys::UINT_VAL_2);
+    $testObj2->example_float->set(ConstKeys::FLOAT_VAL_2);
+
+    return
+      $expected->set(ConstKeys::STRING_KEY_0, $testObj0)
+        ->set(ConstKeys::STRING_KEY_1, $testObj1)
+        ->set(ConstKeys::STRING_KEY_2, $testObj2);
+  }
+
+  public static function getJsonForNestedMap(): string {
```

Comment:
Maybe I missed it, but do we have deeply nested structures in unit tests? eg depth 4 or 5?
And do we support nested structures which combine multiple layers of Storables and TypeInterfaces?